Dataset URL https:/ / data. medicare. gov/ data/ dialysisfacility-compare
This is a simple notebook where we import the dataset in a pandas dataframe using the DBFS in databricks. We then do some initial analysis and find the best performing states in terms of dialysis + Which state has the highest number of dialysis center

In [2]:
# please make sure that you donot leave header="true", inferSchema="true" while using pandas dataframe within databricks

In [3]:
#Let us import the files as pandas dataframe

In [4]:
import pandas as pd 

df = pd.read_csv("/dbfs/FileStore/tables/ESRD_QIP_Payment_Year_2018.csv")

In [5]:
print('Number of rows: ' + str(df.shape[0])) # returns a tuple containing number of rows and columns
print('Number of columns: ' + str(df.shape[1]))

Number of rows: 6825
Number of columns: 154

In [6]:
df.head()

Facility Name 
 CMS Certification Number (CCN) 
 Alternate CCN 1 
 Address1 
 Address2 
 City 
 State 
 Zip Code 
 Network 
 VAT Catheter Measure Score 
 VAT Catheter Achievement Measure Rate/Ratio 
 Number of Patients Included in VAT Catheter Measure Score Achievement Period 
 VAT Catheter Achievement Period Numerator 
 VAT Catheter Achievement Period Denominator 
 VAT Catheter Improvement Measure Rate/Ratio 
 VAT Catheter Improvement Period Numerator 
 VAT Catheter Improvement Period Denominator 
 VAT Catheter Measure Score Applied 
 National Avg VAT Catheter Measure Score 
 VAT Fistula Measure Score 
 VAT Fistula Achievement Measure Rate/Ratio 
 Number of Patients Included in VAT Fistula Measure Score Achievement Period 
 VAT Fistula Achievement Period Numerator 
 VAT Fistula Achievement Period Denominator 
 VAT Fistula Improvement Measure Rate/Ratio 
 VAT Fistula Improvement Period Numerator 
 VAT Fistula Improvement Period Denominator 
 VAT Fistula Measure Score Applied 
 National Avg Fistula Measure Score 
 Vascular Access Combined Measure Score 
 National Avg Vascular Access Combined Measure Score 
 Kt/V Adult Hemodialysis Measure Score 
 Kt/V Adult Hemodialysis Achievement Measure Rate/Ratio 
 Number of Patients Included in Kt/V Adult Hemodialysis Measure Score Achievement Period 
 Kt/V Adult Hemodialysis Achievement Period Numerator 
 Kt/V Adult Hemodialysis Achievement Period Denominator 
 Kt/V Adult Hemodialysis Improvement Measure Rate/Ratio 
 Kt/V Adult Hemodialysis Improvement Period Numerator 
 Kt/V Adult Hemodialysis Improvement Period Denominator 
 Kt/V Adult Hemodialysis Measure Score Applied 
 ... 
 Mineral Metabolism Measure Score 
 Number of Patients Included in Mineral Metabolism Measure Score Achievement Period 
 National Avg Mineral Metabolism Measure Score 
 Anemia Management Measure Score 
 Number of Patients Included in Anemia Management Measure Score Achievement Period 
 National Avg Anemia Management Measure Score 
 Clinical Depression Screening and Follow-up Measure Score 
 Number of Patients Included in Clinical Depression Screening and Follow-up Measure Score Achievement Period 
 National Avg Clinical Depression Screening and Follow-up Measure Score 
 NHSN Influenza Measure Score 
 National Avg NHSN Influenza Measure Score 
 Pain Assessment and Follow-up Measure Score 
 Number of Patients Included in Pain Assessment and Follow-up Measure Score Achievement Period 
 National Avg Pain Assessment and Follow-up Measure Score 
 SRR Measure Score 
 SRR Achievement Measure Rate/Ratio 
 SRR Index Discharges included in Achievement Period 
 SRR Achievement Period Numerator 
 SRR Achievement Period Denominator 
 SRR Improvement Measure Rate/Ratio 
 SRR Improvement Period Numerator 
 SRR Improvement Period Denominator 
 SRR Measure Score Applied 
 National Avg SRR Measure Score 
 STrR Measure Score 
 STrR Achievement Measure Rate/Ratio 
 STrR Patient-Years at Risk 
 STrR Achievement Period Numerator 
 STrR Achievement Period Denominator 
 STrR Improvement Measure Rate/Ratio 
 STrR Improvement Period Numerator 
 STrR Improvement Period Denominator 
 STrR Measure Score Applied 
 National Avg STrR Measure Score 
 Total Performance Score 
 PY2018 Payment Reduction Percentage 
 CMS Certification Date 
 Ownership as of December 31, 2016 
 Date of Ownership Record Update 
 Location 1 
 
 
 
 
 0 
 CHILDRENS HOSPITAL DIALYSIS 
 12306 
 013300 
 1600 7TH AVENUE SOUTH 
 - 
 BIRMINGHAM 
 AL 
 35233 
 8 
 No Score 
 No Rate 
 <11 
 NaN 
 NaN 
 No Rate 
 NaN 
 NaN 
 NaN 
 5 
 No Score 
 No Rate 
 <11 
 NaN 
 NaN 
 No Rate 
 NaN 
 NaN 
 NaN 
 5 
 No Score 
 5 
 No Score 
 No Rate 
 <11 
 NaN 
 NaN 
 No Rate 
 NaN 
 NaN 
 NaN 
 ... 
 10 
 21 
 9 
 10 
 21 
 10 
 10 
 15 
 10 
 10 
 10 
 No Score 
 <11 
 10 
 6 
 0.944 
 31 
 11.0 
 11.156 
 1.642 
 15.0 
 9.137 
 Improvement 
 5 
 No Score 
 No Ratio 
 <10 
 NaN 
 NaN 
 No Ratio 
 NaN 
 NaN 
 NaN 
 5 
 70 
 No Reduction 
 11/17/1982 
 NOT A CHAIN 
 03/06/2012 
 NaN 
 
 
 

In [7]:
print(df.head(n=3)) # n=5 is the number of rows to be displayed

In [8]:
print(df.columns)

Index(['Facility Name', 'CMS Certification Number (CCN)', 'Alternate CCN 1',
 'Address1', 'Address2', 'City', 'State', 'Zip Code', 'Network',
 'VAT Catheter Measure Score',
 ...
 'STrR Improvement Period Numerator',
 'STrR Improvement Period Denominator', 'STrR Measure Score Applied',
 'National Avg STrR Measure Score', 'Total Performance Score',
 'PY2018 Payment Reduction Percentage', 'CMS Certification Date',
 'Ownership as of December 31, 2016', 'Date of Ownership Record Update',
 'Location 1'],
 dtype='object', length=154)

In [9]:
for column in df.columns:
    print(column) # Pandas abbreviates the output so to see all columns name we loop it

Facility Name
CMS Certification Number (CCN)
Alternate CCN 1
Address1
Address2
City
State
Zip Code
Network
VAT Catheter Measure Score
VAT Catheter Achievement Measure Rate/Ratio
Number of Patients Included in VAT Catheter Measure Score Achievement Period
VAT Catheter Achievement Period Numerator
VAT Catheter Achievement Period Denominator
VAT Catheter Improvement Measure Rate/Ratio
VAT Catheter Improvement Period Numerator
VAT Catheter Improvement Period Denominator
VAT Catheter Measure Score Applied
National Avg VAT Catheter Measure Score
VAT Fistula Measure Score
VAT Fistula Achievement Measure Rate/Ratio
Number of Patients Included in VAT Fistula Measure Score Achievement Period
VAT Fistula Achievement Period Numerator
VAT Fistula Achievement Period Denominator
VAT Fistula Improvement Measure Rate/Ratio
VAT Fistula Improvement Period Numerator
VAT Fistula Improvement Period Denominator
VAT Fistula Measure Score Applied
National Avg Fistula Measure Score
Vascular Access Combined Measure Score
National Avg Vascular Access Combined Measure Score
Kt/V Adult Hemodialysis Measure Score
Kt/V Adult Hemodialysis Achievement Measure Rate/Ratio
Number of Patients Included in Kt/V Adult Hemodialysis Measure Score Achievement Period
Kt/V Adult Hemodialysis Achievement Period Numerator
Kt/V Adult Hemodialysis Achievement Period Denominator
Kt/V Adult Hemodialysis Improvement Measure Rate/Ratio
Kt/V Adult Hemodialysis Improvement Period Numerator
Kt/V Adult Hemodialysis Improvement Period Denominator
Kt/V Adult Hemodialysis Measure Score Applied
National Avg Kt/V Adult Hemodialysis Measure Score
Kt/V Adult Peritoneal Dialysis Measure Score
Kt/V Adult Peritoneal Dialysis Achievement Measure Rate/Ratio
Number of Patients Included in Kt/V Adult Peritoneal Dialysis Measure Score Achievement Period
Kt/V Adult Peritoneal Dialysis Achievement Period Numerator
Kt/V Adult Peritoneal Dialysis Achievement Period Denominator
Kt/V Adult Peritoneal Dialysis Improvement Measure Rate/Ratio
Kt/V Adult Peritoneal Dialysis Improvement Period Numerator
Kt/V Adult Peritoneal Dialysis Improvement Period Denominator
Kt/V Adult Peritoneal Dialysis Measure Score Applied
National Avg Kt/V Adult Peritoneal Dialysis Measure Score
Kt/V Pediatric Hemodialysis Measure Score
Kt/V Pediatric Hemodialysis Achievement Measure Rate/Ratio
Number of Patients Included in Kt/V Pediatric Hemodialysis Measure Score Achievement Period
Kt/V Pediatric Hemodialysis Achievement Period Numerator
Kt/V Pediatric Hemodialysis Achievement Period Denominator
Kt/V Pediatric Hemodialysis Improvement Measure Rate/Ratio
Kt/V Pediatric Hemodialysis Improvement Period Numerator
Kt/V Pediatric Hemodialysis Improvement Period Denominator
Kt/V Pediatric Hemodialysis Measure Score Applied
National Avg Kt/V Pediatric Hemodialysis Measure Score
Kt/V Pediatric Peritoneal Dialysis Measure Score
Kt/V Pediatric Peritoneal Dialysis Achievement Measure Rate/Ratio
Number of Patients Included in Kt/V Pediatric Peritoneal Dialysis Measure Score Achievement Period
Kt/V Pediatric Peritoneal Dialysis Achievement Period Numerator
Kt/V Pediatric Peritoneal Dialysis Achievement Period Denominator
Kt/V Pediatric Peritoneal Dialysis Improvement Measure Rate/Ratio
Kt/V Pediatric Peritoneal Dialysis Improvement Period Numerator
Kt/V Pediatric Peritoneal Dialysis Improvement Period Denominator
Kt/V Pediatric Peritoneal Dialysis Measure Score Applied
National Avg Kt/V Pediatric Peritoneal Dialysis Measure Score
Kt/V Dialysis Adequacy Combined Measure Score
National Avg Kt/V Dialysis Adequacy Combined Measure Score
Hypercalcemia Measure Score
Hypercalcemia Achievement Measure Rate/Ratio
Number of Patients Included in Hypercalcemia Measure Score Achievement Period
Hypercalcemia Achievement Period Numerator
Hypercalcemia Achievement Period Denominator
Hypercalcemia Improvement Measure Rate/Ratio
Hypercalcemia Improvement Period Numerator
Hypercalcemia Improvement Period Denominator
Hypercalcemia Measure Score Applied
National 

In [10]:
df_states = df.groupby('State').size()
print(df_states)

State
AK 9
AL 170
AR 69
AS 1
AZ 120
CA 625
CO 75
CT 49
DC 23
DE 27
FL 433
GA 345
GU 4
HI 27
IA 70
ID 29
IL 299
IN 161
KS 54
KY 115
LA 169
MA 83
MD 162
ME 17
MI 211
MN 120
MO 162
MP 2
MS 87
MT 13
NC 211
ND 16
NE 37
NH 18
NJ 168
NM 49
NV 46
NY 274
OH 314
OK 84
OR 67
PA 294
PR 45
RI 15
SC 145
SD 25
TN 182
TX 605
UT 41
VA 178
VI 5
VT 8
WA 93
WI 123
WV 42
WY 9
dtype: int64

In [11]:
df_states = df.groupby('State').size().sort_values(ascending=False)
print(df_states) # Count dialysis facility in each state

State
CA 625
TX 605
FL 433
GA 345
OH 314
IL 299
PA 294
NY 274
NC 211
MI 211
TN 182
VA 178
AL 170
LA 169
NJ 168
MD 162
MO 162
IN 161
SC 145
WI 123
AZ 120
MN 120
KY 115
WA 93
MS 87
OK 84
MA 83
CO 75
IA 70
AR 69
OR 67
KS 54
CT 49
NM 49
NV 46
PR 45
WV 42
UT 41
NE 37
ID 29
HI 27
DE 27
SD 25
DC 23
NH 18
ME 17
ND 16
RI 15
MT 13
WY 9
AK 9
VT 8
VI 5
GU 4
MP 2
AS 1
dtype: int64

In [12]:
df_states = df.groupby('State').size().sort_values(ascending=False).head(n=10)
print(df_states) # Sort in descending order & limit output to 10

State
CA 625
TX 605
FL 433
GA 345
OH 314
IL 299
PA 294
NY 274
NC 211
MI 211
dtype: int64

# Observation: Califormia (CA) has the highest number of dialysis centers

In [14]:
df_ca = df.loc[df['State'] == 'CA']
print(df_ca) # Filter centers by State equals California

Facility Name ... Location 1
368 SANTA CLARA VALLEY RENAL CARE CENTER ... NaN
369 CHILDRENS HOSPITAL OF LOS ANGELES ... NaN
370 KAISER FOUNDATION HOSPITAL MEDICAL CTR.- SUNSET ... NaN
371 ARROWHEAD REGIONAL MEDICAL CENTER ... NaN
372 LOMA LINDA UNIVERSITY MEDICAL CENTER ... NaN
373 ST. JOSEPH HOSPITAL RENAL CENTER ... NaN
374 KAISER FOUNDATION HOSPITAL MEDICAL CTR.- FONTANA ... NaN
375 KAISER FOUNDATION HOSPITAL MEDICAL CTR.- WOODL... ... NaN
376 KAISER FOUNDATION HOSPITAL MEDICAL CTR.- RIVER... ... NaN
377 KAISER FOUNDATION HOSPITAL MEDICAL CTR.- WEST ... ... NaN
378 KAISER FOUNDATION HOSPITAL MEDICAL CTR.- DOWNEY ... NaN
379 LUCILE PACKARD STANFORD PEDIATRIC DIALYSIS ... NaN
380 KAISER PERMANENTE - SANTA CLARA ... NaN
381 KAISER FOUNDATION HOSPITAL MEDICAL CTR.- ANAHEIM ... NaN
382 CHILDRENS HOSPITAL OF SAN DIEGO ... NaN
383 KAISER HOSPITAL ESRD PD UNIT ... NaN
384 KAISER FOUNDATION HOSPITAL MEDICAL CTR.- BALDW... ... NaN
385 CHILDRENS HOSPITAL CENTRAL CALIFORNIA ... NaN
386 SIERRA VIEW DISTRICT HOSPITAL DIALYSIS CENTER ... NaN
387 KAISER PERMANENTE MARTINEZ ... NaN
388 KAISER FOUNDATION HOSPITAL MEDICAL CTR.- ONTARIO ... NaN
389 UC DAVIS MEDICAL CENTER PEDIATRIC ... NaN
390 FMC-HILLCREST DIALYSIS ... NaN
391 DAVITA-KIDNEY DIALYSIS CARE UNIT ... NaN
392 DAVITA-MAIN PLACE DIALYSIS CENTER ... NaN
393 DAVITA-FULLERTON DIALYSIS ... NaN
394 RAI HAIGHT ... NaN
395 FMC-SANTA BARBARA COMMUNITY DIALYSIS CENTER ... NaN
396 SATELLITE DIALYSIS STEVENS CREEK ... NaN
397 LOS ALAMITOS HEMODIALYSIS CENTER ... NaN
... ... ... ...
6373 ARCHWAY MODESTO HOME TRAINING ... NaN
6374 DAVITA-ANAHEIM SPRINGS DIALYSIS ... NaN
6375 SATELLITE HEALTHCARE SAN JOSE ... NaN
6376 INNOVATIVE DIALYSIS OF PATTERSON, LLC ... NaN
6377 DAVITA-BAKERSFIELD OAK ST DIALYSIS ... NaN
6378 DESERT CITIES DIALYSIS-AMETHYST ... NaN
6379 DAVITA ROSEVILLE DIALYSIS CENTER ... NaN
6380 WALNUT CREEK WEST DIALYSIS CENTER ... NaN
6381 DAVITA-BLUFF RD DIALYSIS ... NaN
6382 DAVITA-POMONA VALLEY DIALYSIS ... NaN
6383 FMC-SANTA ANA DIALYSIS ... NaN
6384 DAVITA-SAN BERNARDINO HOME TRAINING ... NaN
6385 LAUREL CANYON DIALYSIS, LLC ... NaN
6386 SYLMAR DIALYSIS, LLC ... NaN
6387 DAVITA EL SOBRANTE ... NaN
6388 DAVITA-MENIFEE HOME DIALYSIS ... NaN
6389 DAVITA-GARDEN GROVE HARBOR DIALYSIS ... NaN
6390 US RENAL CARE BEVERLY HILLS DIALYSIS ... NaN
6391 U.S. RENAL CARE SOUTH BAY HOME PROGRAM ... NaN
6392 DAVITA-GLENDORA FOOTHILLS DIALYSIS ... NaN
6393 FRESENIUS MEDICAL CARE ANAHEIM DIALYSIS ... NaN
6394 DAVITA-ARCADIA OAKS DIALYSIS ... NaN
6395 ROSEVILLE HOME THERAPY ... NaN
6396 SANDERLING DIALYSIS OF TEHACHAPI ... NaN
6397 DAVITA-COLTON RANCH DIALYSIS ... NaN
6398 DAVITA-AVALON DIALYSIS ... NaN
6399 KERMAN DIALYSIS CENTER, LLC ... NaN
6400 FMC-SOUTH BAY HOME THERAPY ... NaN
6401 DAVITA-BEVERLYWOOD DIALYSIS ... NaN
6402 DAVITA-EL DORADO DIALYSIS ... NaN

[625 rows x 154 columns]

In [15]:
print(df.groupby('Total Performance Score').size()) # Get a count on the different scores received by centers

Total Performance Score
10 10
100 30
11 2
12 2
13 1
14 3
15 1
17 2
18 2
19 3
20 7
21 5
22 5
23 6
24 5
25 13
26 8
27 9
28 6
29 8
30 9
31 9
32 14
33 17
34 24
35 22
36 29
37 24
38 30
39 33
 ... 
73 159
74 146
75 108
76 104
77 104
78 87
79 95
8 2
80 61
81 72
82 47
83 53
84 48
85 45
86 38
87 25
88 31
89 22
9 1
90 21
91 18
92 24
93 11
94 7
95 15
96 2
97 11
98 8
99 12
No Score 276
Length: 95, dtype: int64

# Observations: We note that there are 276 rows with no values for Total performance score. 
# Column Performance score is a string rather than a integer

In [17]:
import warnings
warnings.filterwarnings('ignore')

In [18]:
# Convert and display 5 results
df_filt= df.loc[df['Total Performance Score'] != 'No Score']
df_filt['Total Performance Score'] = pd.to_numeric(df_filt['Total Performance Score'])
df_tps = df_filt[['Facility Name','State', 'Total Performance Score']].sort_values('Total Performance Score')
print(df_tps.head(n=5))

Facility Name State Total Performance Score
5626 462320 PRIMARY CHILDREN'S DIALYSIS CENTER UT 5
705 PEDIATRIC DIALYSIS UNIT AT UCSF CA 7
6766 VITAL LIFE DIALYSIS CENTER FL 8
4637 BELMONT COURT DIALYSIS - DOYLESTOWN CAMPUS PA 8
3768 WOODMERE DIALYSIS LLC NY 9

In [19]:
import numpy as np

df_state_means = df_filt.groupby('State').agg({'Total Performance Score': np.mean})
print(df_state_means.sort_values('Total Performance Score', ascending=False))

Total Performance Score
State 
ID 73.178571
WY 71.777778
HI 70.500000
UT 70.421053
CO 70.173333
WA 70.146067
ME 70.058824
OR 70.046154
KS 69.480769
AZ 68.905983
NM 68.711111
AK 66.250000
RI 66.200000
NE 66.083333
IN 65.868421
IA 65.590909
MN 65.521368
MO 65.113924
NV 64.888889
NJ 64.654321
CT 64.625000
WI 64.588235
SD 63.708333
KY 63.553571
CA 63.333333
MA 63.192308
AL 62.975758
VA 62.845238
OK 62.722892
PA 62.525000
MT 62.416667
IL 62.302583
NH 62.055556
NC 61.643216
OH 61.614334
GA 61.582583
TX 61.325901
DE 60.653846
MS 60.548780
MI 60.525253
LA 59.766871
WV 59.756757
SC 59.451389
TN 59.153409
VT 57.375000
FL 57.347518
MD 57.047619
AR 56.712121
NY 55.841509
DC 54.333333
ND 53.000000
PR 51.955556
GU 48.250000
VI 43.400000
MP 38.000000
AS 37.000000

# Observations: IDAHO and WYOMING are the best performing dialysis center
# Both are moutain based area, do altitude have a role?

In [21]:
import numpy as np

df_state_means = df_filt.groupby('State').agg({ # Add a column to include the number of dialysis center
    'Total Performance Score': np.mean,
    'State': np.size
})
print(df_state_means.sort_values('Total Performance Score', ascending=False))

Total Performance Score State
State 
ID 73.178571 28
WY 71.777778 9
HI 70.500000 26
UT 70.421053 38
CO 70.173333 75
WA 70.146067 89
ME 70.058824 17
OR 70.046154 65
KS 69.480769 52
AZ 68.905983 117
NM 68.711111 45
AK 66.250000 8
RI 66.200000 15
NE 66.083333 36
IN 65.868421 152
IA 65.590909 66
MN 65.521368 117
MO 65.113924 158
NV 64.888889 45
NJ 64.654321 162
CT 64.625000 48
WI 64.588235 119
SD 63.708333 24
KY 63.553571 112
CA 63.333333 615
MA 63.192308 78
AL 62.975758 165
VA 62.845238 168
OK 62.722892 83
PA 62.525000 280
MT 62.416667 12
IL 62.302583 271
NH 62.055556 18
NC 61.643216 199
OH 61.614334 293
GA 61.582583 333
TX 61.325901 583
DE 60.653846 26
MS 60.548780 82
MI 60.525253 198
LA 59.766871 163
WV 59.756757 37
SC 59.451389 144
TN 59.153409 176
VT 57.375000 8
FL 57.347518 423
MD 57.047619 147
AR 56.712121 66
NY 55.841509 265
DC 54.333333 21
ND 53.000000 15
PR 51.955556 45
GU 48.250000 4
VI 43.400000 5
MP 38.000000 2
AS 37.000000 1